Страны и города

Maksonchik88 opened the pull request:
https://informatics.msk.ru/mod/statements/view.php?id=4535&chapterid=3765#1

*Открыл файл
*Инициализировал словарь, где страна -> города
*Вывел результат, согласнотзаданию
*Забыл закрыть файл

Maksonchik88 committed:
Страны и города

https://informatics.msk.ru/mod/statements/view.php?id=4535&chapterid=3765#1

*Открыл файл
*Инициализировал словарь, где страна -> города
*Вывел результат, согласнотзаданию
*Забыл закрыть файл

iliadmitriev commented on dict/countries_and_cities.py:
```diff
@@ -0,0 +1,18 @@
+        data[country] = cities
+num_2 = int(file.readline())
+
+for _ in range(num_2):
+    line = file.readline().rstrip()
+    for country, cities in data.items():
+        if line in cities:
+            print(country)
```
Словарь data отображает страну в список городов.
Что если поменять словарь и сделать чтобы было отображение город -> страна, то можно обойтись без вложенного цикла и использовать словарь для проверки есть ли в нем нужный ключ.

iliadmitriev reviewed (changes requested):
Поменяй словарь data
Из страна -> список городов
В город -> страна

Maksonchik88 committed:
*Поменял местами города -> страны
*Все равно получается вложенный цикл, только в перво м случае
*Чувствую, что можно сделать с помощью fromkeys(), но не понимаю как

iliadmitriev commented on dict/countries_and_cities.py:
```diff
@@ -0,0 +1,19 @@
+    cities = line[1:]
+    for city in cities:
+        data[city] = country
+print(data)
```
Отладочный принт

iliadmitriev reviewed (approved):
Другое дело. Код выглядит гораздо проще. 👍
